lecture_notes/09 functions/2-scope.py

```diff
@@ -0,0 +1,27 @@
+cities = ['Athens', 'Roma', 'London', 'Beijing', 'Kiev', 'Riga']
+def f(s):
+    a = len(s)
+    return a
+def f1():
+    cities = ['Athens', 'Roma']
+    def f(s):
+        a = s.upper()
+        return a
+    print(cities)
+    print(list(map(f, cities)))
+    def f2():
+        def f(s):
+            a = s.lower()
+            return a
+        print(cities)
+        print(list(map(f, cities)))
+    def f3():
+        cities = ['London', 'Beijing', 'Kiev']
+        print(cities)
+        print(list(map(f, cities)))
+    f2()
+    f3()
+
+f1()
+print(cities)
+print(list(map(f, cities)))
```

Comment:
This example shows us variable scoping, not lower/upper cities, it's just an example